algviz/interface/visitors.py

```diff
@@ -190,3 +190,29 @@ class WidgetVisitor(Visitor):
 
     def visit(self, *args, **kwargs):
         return super().visit(*args, **kwargs)
+
+class LinkedListVisitor(Visitor):
+    """Linked lists are represented as an array of an array and a current index, i.e.,
+    [[1, 2, 3], 0]."""
```

Comment:
We should use a tuple or a `namedtuple` instead of a `list` of constant size 2.
Also, it'd be best if we automatically converted a `list` argument `l` to the tuple (or namedtuple) `(l, 0)`.  That way, the gimmick where we represent a linked list node as a `(list, int)` pair is encapsulated.